packages/next/src/client/components/router-reducer/fetch-server-response.ts

```diff
@@ -84,6 +87,7 @@ export async function fetchServerResponse(
       headers[NEXT_ROUTER_PREFETCH_HEADER] || '0',
       headers[NEXT_ROUTER_STATE_TREE],
       headers[NEXT_URL],
+      headers[NEXT_BUILD_ID_HEADER],
```

Comment:
If we don't want to send the build id in the Vary header, we can just add the `currentBuildId` here directly, which is what I had originally in https://github.com/vercel/next.js/pull/67229/commits/8f212819dc2d81abc2d7555d5f0bab470b0ce1a9